homer.root/usr/share/clearwater/infrastructure/scripts/homer

```diff
@@ -1,18 +1,61 @@
+
+# @file homer
+#
+# Project Clearwater - IMS in the Cloud
+# Copyright (C) 2013 Metaswitch Networks Ltd
+#
+# This program is free software: you can redistribute it and/or modify it
+# under the terms of the GNU General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or (at your
+# option) any later version, along with the "Special Exception" for use of
+# the program along with SSL, set forth below. This program is distributed
+# in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+# without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details. You should have received a copy of the GNU General Public
+# License along with this program.  If not, see
+# <http://www.gnu.org/licenses/>.
+#
+# The author can be reached by email at clearwater@metaswitch.com or by
+# post at Metaswitch Networks Ltd, 100 Church St, Enfield EN2 6BQ, UK
+#
+# Special Exception
+# Metaswitch Networks Ltd  grants you permission to copy, modify,
+# propagate, and distribute a work formed by combining OpenSSL with The
+# Software, or a work derivative of such a combination, even if such
+# copying, modification, propagation, or distribution would otherwise
+# violate the terms of the GPL. You must comply with the GPL in all
+# respects for all of the code used other than OpenSSL.
+# "OpenSSL" means OpenSSL toolkit software distributed by the OpenSSL
+# Project and licensed under the OpenSSL Licenses, or a work based on such
+# software and licensed under the OpenSSL Licenses.
+# "OpenSSL Licenses" means the OpenSSL License and Original SSLeay License
+# under which the OpenSSL Project distributes the OpenSSL toolkit software,
+# as those licenses appear in the file LICENSE-OPENSSL.
+
 # Set default values which will be overwritten if they exist in /etc/clearwater/config
 cassandra_hostname="localhost" 
 . /etc/clearwater/config
-function escape { echo $1 | sed -e 's/\//\\\//g' ; }
-sed -e 's/^LOCAL_IP = .*$/LOCAL_IP = "'$(escape $local_ip)'"/g
-        s/^SIP_DIGEST_REALM = .*$/SIP_DIGEST_REALM = "'$(escape $home_domain)'"/g
-        s/^SPROUT_HOSTNAME = .*$/SPROUT_HOSTNAME = "'$(escape $sprout_hostname)'"/g
-        s/^CASS_HOST = .*$/CASS_HOST = "'$(escape $cassandra_hostname)'"/g' </usr/share/clearwater/homer/src/metaswitch/crest/local_settings.py >/tmp/local_settings.py.$$
-for dst in /usr/share/clearwater/homer/src/metaswitch/crest/local_settings.py \
-           /usr/share/clearwater/homer/env/lib/python2.7/site-packages/crest-0.1-py2.7.egg/metaswitch/crest/local_settings.py
-do
-  if [ -f $dst ]
-  then
-    cp /tmp/local_settings.py.$$ $dst
-  fi
-done
-rm /tmp/local_settings.py.$$
+
+# Only modify local_settings if the home_domain, sprout_hostname and local_ip
+# are available (they might not be if shared_config hasn't been propagated
+# round the deployment yet). Homer will cyclically restart until this has
```

Comment:
The local settings file starts with these entries set to MUST_BE_CONFIGURED. Homer exits on this